bin/main.rb

```diff
@@ -1,91 +1,105 @@
-
-puts "#{player_2_name}, this cell is already taken, choose again!"
-print "\n#{player_2_name} select cell between 1-9\n" # Player selects cell 4
-puts
-puts 'X | 2 | 3 '
-puts '----------'
-puts 'O | 5 | 6 '
-puts '----------'
-puts '7 | 8 | 9 '
-
-print "\n#{player_1_name} select cell between 1-9\n" # Assuming he selects cell 2
-puts
-
-puts 'X | X | 3 '
-puts '----------'
-puts 'O | 5 | 6 '
-puts '----------'
-puts '7 | 8 | 9 '
-
-print "\n#{player_2_name} select cell between 1-9\n" # the game continues as intended
-puts
-# assuming Player-2 selects cell 8, the output is displayed next
-puts 'X | X | 3 '
-puts '----------'
-puts 'O | 5 | 6 '
-puts '----------'
-puts '7 | O | 9 '
-
-print "\n#{player_1_name} select cell between 1-9\n" # the game continues as intended
-puts
-# assuming Player-1 selects cell 3, the output is displayed next
-puts 'X | X | X '
-puts '----------'
-puts 'O | 5 | 6 '
-puts '----------'
-puts '7 | O | 9 '
-
-# end
-# if player_1 has his symbols in an order that matches a winning output game_on = false and the loop should break
-print "\n#{player_1_name} wins!!"
+puts "\t X | O | X"
+puts "\t---|---|---"
+puts "\t O | X | O"
+puts "\t---|---|---"
+puts "\t X | O | X"
 puts
-# At this point, the game asks the players if they'll like to play again, if yes, they'll press enter to start again.
-# If not then they'll exit the game.
 
-# if no player has a winning output, but the board is filled with their symbols, game_on = false and the loop will break
-# and it will output game is a draw and ask if they want to play again.
+game = true
+
+while game
+  print "Player One press \'P'\ to enter name or \'Enter'\ to play as guest: "
+  id = gets.chomp.upcase
+  if id == 'P'
+    print 'Player One name please: '
+    players1 = gets.chomp
+  else
+    players1 = 'Guest__01'
+  end
+  puts
+  print "Player Two press \'P'\ to enter name or \'Enter'\ to play as guest: "
+  id2 = gets.chomp.upcase
+  if id2 == 'P'
+    print 'Player Two name please: '
+    players2 = gets.chomp
+  else
+    players2 = 'Guest__02'
+  end
+  puts
+
+  player1 = Players.new(players1, 'X')
+  player2 = Players.new(players2, 'O')
+  puts "#{player1.name} your symbol is #{player1.char}"
+  puts
+  puts "#{player2.name} your symbol is #{player2.char}"
+  puts
+  puts 'This is your playing board'
+  board = Board.new
+  puts board.display_board
+  puts
+
+  playerz = [player1, player2]
+
+  game_on = true
+
+  while game_on
+    playerz.each do |playa|
+      if game_on
+
+        loop do
+          print "#{playa.name} choose a cell: "
+          input = gets.chomp.to_i
+          inputs = input - 1
+
+          if board.positions.include?(input)
+            playa.player_input << inputs
+            board.positions[inputs] = playa.char
+            break
+          else
+            puts 'Invalid move/character try again!'
+            puts
+            next
+          end
```

Comment:
1. **Check** the input validity into a method from `Board` class
```ruby
def input_valid?(input)
...
return true/false
end
```

2. **Update the board** into the method from a `Board` class
```ruby
def update_board
@positions[inputs] = current_player.char
end
```